### PR TITLE
Use latest version of the Contributor Covenant.

### PR DESCRIPTION
**Pull request by ossama-othman**
Update the mptcpd Code of Conduct to the latest version of the
Contributor Covenant, i.e. v2.0.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,76 +1,134 @@
...
+reported to the community leaders responsible for enforcement at
+mptcp-owner@lists.01.org.
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
```
Comment (mjmartineau) [inline CODE_OF_CONDUCT.md:72]: The header above uses "Enforcement Guidelines", and "Community Impact Guidelines" is on this line. Should there be one name for the guidelines?

**Review by mjmartineau — APPROVED** [review]
Overall, the changes look good and I like the upstream improvements in the 2.0 covenant.

One comment about the naming of the guideline section, see inline. I'm flexible w.r.t. merging as-is or making the edit to the guideline naming.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (ossama-othman) [reply CODE_OF_CONDUCT.md:72]: That sounds reasonable but we would diverge from [upstream](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) by making such a change.  Other than filling in the reporting e-mail address placeholder I'm reluctant to diverge from upstream.

Comment (mjmartineau) [reply CODE_OF_CONDUCT.md:72]: Oh, my mistake. I thought you had brought in the Mozilla-inspired code-of-conduct ladder yourself for some reason. Better to stay with the 2.0 text!

**Pull request merged**